src/rust/ide/view/graph-editor/src/component/visualization/foreign/java_script/binding.rs

```diff
@@ -86,10 +94,57 @@ impl JsConsArgs {
         self.root.clone().into()
     }
 
+    /// Getter for the theming API that we expose to JS visualizations
+    pub fn theme(&self) -> JsTheme {
+        self.theme.clone()
+    }
+
     /// Helper method to emit an preprocessor change event from the visualisation.
     pub fn emit_preprocessor_change(&self, code:Option<String>, module:Option<String>){
         let closure             = &self.set_preprocessor;
         let preprocessor_config = PreprocessorConfiguration::from_options(code,module);
         (*closure)(preprocessor_config);
     }
 }
+
+/// The theming API that we expose to JS visualizations
+#[wasm_bindgen]
+#[derive(Clone, Debug)]
+pub struct JsTheme {
+    styles: StyleWatch
+}
+
+#[allow(non_snake_case)]
+#[wasm_bindgen]
+impl JsTheme {
+    /// Takes a qualified type name and returns the color that is used in the GUI for that type.
+    pub fn getColorForType(&self, tp_name: &str) -> String {
+        let tp = Type::from(tp_name.to_string());
+        let lcha = type_coloring::compute(&tp,&self.styles);
+        format_lcha(lcha)
+    }
+
+    /// Takes a qualified type name and returns the color that should be used for foreground
+    /// (e.g. text) that is shown on top of the background color returned by getColorForType.
+    pub fn getForegroundColorForType(&self, _tp_name: &str) -> String {
+        "white".to_string()
+    }
+
+    /// Queries style sheet value for a value.
+    pub fn get(&self, path: &str) -> Option<String> {
+        if let style::Data::Color(lcha) = self.styles.get(path)? {
+            Some(format_lcha(lcha))
+        } else {
+            None
+        }
+    }
+}
+
+fn format_lcha(lcha: color::Lcha) -> String {
+    let rgba = color::Rgba::from(lcha);
+    format!("rgba({:.0}, {:.0}, {:.0}, {})",
+        (rgba.red * 255.0).round(),
+        (rgba.green * 255.0).round(),
+        (rgba.blue * 255.0).round(),
+        rgba.alpha)
+}
```

Comment:
could you add method `to_javascript_string` to both `Rgb` and `Rgba` instead, please?

Reply:
I would actually suggest that we get these values as an array of 4 numbers (RGBA) instead of a string.

For example, in the DB visualization we need to change the alpha to use for the background. If it is a string, it would get complicated and require parsing, while an array can be easily modified and then rendered into a desired format.

Reply:
@radeusgd and I decided to pass the raw numbers to the visualisation, so this should be obsolete.